### PR TITLE
Change string to DU

### PR DESCRIPTION
**Pull request by awf**
When viewing this sample, I thought it might be nice to remove the "_" match on (old) line 176.  To do so, I thought it might be nicer to use a discriminated union instead of strings to indicate state.  Hence this PR -- I hope you like it :)  All comments very much welcome -- I am a distinct newbie.

### CODE REVIEW

File: public/samples/elmish/todomvc.fs [hunk 1]
```diff
@@ -17,9 +17,17 @@ open Elmish.React
...
+type WhatIsVisible =
+   | All 
+   | Active
+   | Completed 
+
+let toStr v = 
+    match v with
+    |All -> "All"
+    |Active -> "Active"
+    |Completed -> "Completed"
```
Comment (MangelMaxime) [inline public/samples/elmish/todomvc.fs:30]: Missing a space after `|`

**Review by MangelMaxime — CHANGES_REQUESTED** [review]

File: public/samples/elmish/todomvc.fs (hunk 1 above)
Comment (awf) [reply public/samples/elmish/todomvc.fs]: Fixed, thanks!

**Comment by awf** [comment]
@cgravill called this "moving from string typing to strong typing", which I found very amusing.

**Comment by cgravill** [comment]
I've found a lot of incidental programmers much happier with F#/Fable (than JavaScript) once we started using more typing features. Combined with the excellent Elm(ish) approach it lets them concentrate on mapping domain types to a visualisation without worrying they're breaking the infrastructure programming.

**Comment by alfonsogarciacaro** [comment]
Sorry, missed the notification! This is a good change, I think the string remained from the original React example we converted. Thank you!

**Comment by awf** [comment]
Thanks!

**Pull request merged**